lib/mongoid/threaded.rb

```diff
@@ -197,27 +197,37 @@ def client_override=(name)
     # Get the current Mongoid scope.
     #
     # @example Get the scope.
-    #   Threaded.current_scope
+    #   Threaded.current_scope(klass)
     #
     # @return [ Criteria ] The scope.
     #
     # @since 5.0.0
-    def current_scope
-      Thread.current[CURRENT_SCOPE_KEY]
+    def current_scope(klass)
```

Comment:
We should have the argument be (klass = nil) so that we don't break the API. This allows you to call the method as before 
```ruby
Threaded.current_scope
```
without an argument and no argument error will be thrown. This is kind of like overloading, though not exactly.

Then in the method body, you should check if klass is nil. If so, you return Thread.current[CURRENT_SCOPE_KEY], otherwise you can run the code you've written for finding the scope by class.